src/pages/MovieDetail/MovieDetail.js

```diff
@@ -1,17 +1,188 @@
 import React from 'react';
 import MovieBannerSection from './Components/MovieBannerSection/MovieBannerSection';
 import LeaveCommentSection from './Components/LeaveCommentSection/LeaveCommentSection';
+import ShowComment from './Components/LeaveCommentSection/ShowComment';
 import MovieDetailContentsSection from './Components/MovieDetailContentsSection/MovieDetailContentsSection';
+import API_URLS from '../../config';
 
 import './MovieDetail.scss';
 
 export default class MovieDetail extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userWish: false,
+      leaveComment: false,
+      showComment: false,
+      commentModal: false,
+      commentInputValue: '',
+      movieInformation: {},
+      comment_id: 0,
+    };
+  }
+
+  changeStateOfWish = () => {
+    const movieId = this.props.match.params.id;
+    let token = localStorage.getItem('TOKEN') || '';
+    fetch(`${API_URLS.DETAIL}/${movieId}/wish`, {
+      method: 'POST',
+      headers: {
+        Authorization: token,
+      },
+      body: JSON.stringify({ movie_id: movieId }),
+    }).then(res => {
+      if (res.status === 201) {
+        this.setState({ userWish: true, leaveComment: true });
+      } else if (res.status === 204) {
+        this.setState({
+          userWish: false,
+          leaveComment: false,
+          showComment: false,
+        });
+        return;
+      } else if (res.status === 401) {
+        alert('로그인이 필요합니다!');
+      }
```

Comment:
```suggestion
      if (res.status === 201) {
         return this.setState({ userWish: true, leaveComment: true });
      }
      if (res.status === 204) {
        return this.setState({
          userWish: false,
          leaveComment: false,
          showComment: false,
        });
      }
      if (res.status === 401) {
        return alert('로그인이 필요합니다!');
      }
```
이 좀 더 깔끔하고 적합해보입니다!
아니라면 switch문을 적용하시는 것도 괜찮아보이고요